server/src/main/java/org/candlepin/async/JobManager.java

```diff
@@ -1096,8 +1084,7 @@ public AsyncJobStatus queueJob(JobConfig config) throws JobException {
             // If this occurs do to some other unexpected failure, we'll have some state cleanup
             // to deal with, probably.
 
-            this.setJobState(status, JobState.FAILED);
-            status.setJobResult(e.toString());
+            throw new JobException(e, true);
```

Comment:
In both of those cases, we have a job that has already been persisted, and the failure is more likely from being unable to send an artemis message. If it is a DB error, the exception on update will be rethrown (as you noted) and that will propagate through the system and fail in the expected way.

Reply:
That's fine, but we still have another problem because in the first case, the state that was persisted was CREATED, and we can't move it from CREATED to FAILED (invalid transition). And if we did change that to ABORTED, then the 2nd case wouldn't work because FAILED_WITH_RETRY -> ABORTED is invalid. So maybe we should add ABORTED to the valid transitions from FAILED_WITH_RETRY? (alternatively, we could simply make CREATED -> FAILED valid, but I would prefer the other solution).

Reply:
Ahh, gotcha. I've added the FAILED_WITH_RETRY -> ABORTED transition, and updated the exception handler in message posting to use ABORTED.